### PR TITLE
Add KDoc

### PR DESCRIPTION
**Pull request by cmsigrist**
Add documentation to the code I've written

### CODE REVIEW

**Review by cgntnr — APPROVED** [review]
Good. I can only see documenting and refactoring yet the coverage seems to be affected negatively. Any idea why?

**Comment by cmsigrist** [comment]
> Good. I can only see documenting and refactoring yet the coverage seems to be affected negatively. Any idea why?

Yes I saw and I really have no idea why, since I only added doc to the User... and for some reason ProfilePassion went from B to A........ thank you codeclimate for being very insightful x)

File: app/src/main/java/ch/epfl/sdp/blindly/ViewModelFactory.kt [hunk 1]
```diff
@@ -12,16 +12,28 @@ import ch.epfl.sdp.blindly.user.UserRepository
 import dagger.assisted.Assisted
 import dagger.assisted.AssistedInject
 
+/**
+ * A factory that help to create ViewModels that uses the UserRepository
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/ViewModelFactory.kt:16]:
```suggestion
 * A factory that helps to create ViewModels that uses the UserRepository
```

File: app/src/main/java/ch/epfl/sdp/blindly/main_screen/MessagePageFragment.kt [hunk 2]
```diff
@@ -10,11 +10,16 @@ import dagger.hilt.android.AndroidEntryPoint
...
 class MessagePageFragment : Fragment() {
-
     companion object {
         private const val ARG_COUNT = "messageArgs"
         private var counter: Int? = null
 
+        /**
+         * Create a new instance of MessagePageFragment
+         *
+         * @param counter the position of the framgnet in the TabLayout
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/main_screen/MessagePageFragment.kt:20]:
```suggestion
         * @param counter the position of the fragment in the TabLayout
```

File: app/src/main/java/ch/epfl/sdp/blindly/main_screen/profile/ProfilePageFragment.kt [hunk 3]
```diff
@@ -102,8 +115,9 @@ class ProfilePageFragment : Fragment() {
 
     /**
      * An onClickListener that start an Activity after the button has stopped bouncing
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/main_screen/profile/ProfilePageFragment.kt:117]:
```suggestion
     * An onClickListener that starts an Activity after the button has stopped bouncing
```

File: app/src/main/java/ch/epfl/sdp/blindly/user/User.kt [hunk 4]
```diff
@@ -40,51 +55,117 @@ class User private constructor(
...
+         *
+         * @param description the description of the User
+         */
         fun setDescription(description: String) = apply {
             this.description = description
         }
 
-        fun build() : User {
+        /**
+         * Build a User from the UserBuidler parameters
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/user/User.kt:152]:
```suggestion
         * Build a User from the UserBuilder parameters
```

File: app/src/main/java/ch/epfl/sdp/blindly/user/UserCache.kt [hunk 5]
```diff
@@ -2,27 +2,41 @@ package ch.epfl.sdp.blindly.user
...
+     * @return The User if it was cached, null otherwise
+     */
     @RequiresApi(Build.VERSION_CODES.N)
-    fun get(uid:String): User? {
+    fun get(uid: String): User? {
         return cache[uid]
     }
 
+    /**
+     * Put the given the User in the cache
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/user/UserCache.kt:25]:
```suggestion
     * Put the given User in the cache
```

File: app/src/main/java/ch/epfl/sdp/blindly/user/UserHelper.kt [hunk 6]
```diff
@@ -139,16 +139,21 @@ class UserHelper {
         return FirebaseAuth.getInstance().currentUser?.email
     }
 
+    /**
+     * Enable the user to change his password
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/user/UserHelper.kt:143]:
```suggestion
     * Enable the user to change their password
```

**Review by Lollipoke — APPROVED** [review]
Some small details / typos but overall great documentation :)

**Pull request merged**